src/Security/ContaoAuthenticator.php

```diff
@@ -28,6 +30,21 @@
 class ContaoAuthenticator implements SimplePreAuthenticatorInterface
 {
     /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * Constructor.
+     *
+     * @param ContainerInterface $container The service container
+     */
+    public function __construct(ContainerInterface $container)
```

Comment:
We should simply extend `ContainerAware` and handle the case `null === $this->container`. Problem solved without API change :)